### ISSUE TITLE
Gossip custom message data

### ISSUE THREAD

**Issue by edwardcapriolo** [edwardcapriolo/gossip#7]
We can add a field to the message format that can be used to gossip custom data. We can have an extra field that can be a hashmap. Users can serialize anything into the map.

**Comment by hvandenb** [comment]
This is will be a good idea. Similar Akka is doing this.

**Comment by edwardcapriolo** [comment]
Yes thinking through this though I am not sure if the gossiper is the right place to exchange data. Where does the gossiper and the database/service using the gossiper begin?

**Comment by hvandenb** [comment]
Agreed that gossip is not the right protocol to exchange data. This would put additional constraints on the protocol to deal with things like consistency etc. However, there might be a good to provide additional node meta data that a client wants to exchange. 

Akka adds version to do some optimization for version https://gist.github.com/jboner/7692270

**Comment by edwardcapriolo** [comment]
One of the reasons I found/put together this library was I was building my own no-sql https://github.com/edwardcapriolo/nibiru. So I was thinking the place to do things like this would be in there.

**Comment by hvandenb** [comment]
Are you thinking about continuing this library, i.e. gossip?

**Comment by edwardcapriolo** [comment]
Yes. We can go on with the library. But I *think* a eventually consistent CRDT structure is something closer to a NoSQL database than a gossip library.

**Comment by hvandenb** [comment]
for sure and that would not make sense to add to this library.
